h2o-py/tests/testdir_algos/stackedensemble/pyunit_stackedensemble_base_models.py

```diff
@@ -109,8 +111,94 @@ def test_base_models_are_populated():
     assert pu.is_type(retrieved_se.base_models, [str])
 
 
+# PUBDEV-4534
+def test_stacked_ensemble_accepts_mixed_definition_of_base_models():
+    """This test asserts that base models can be one of these:
+    * list of models
+    * GridSearch
+    * list of GridSearches
+    * list of Gridsearches and models
+    """
+    hyper_parameters = dict()
+    hyper_parameters["ntrees"] = [1, 3, 5]
+
+    data = prepare_data()
+
+    drf = H2ORandomForestEstimator(
+        fold_assignment="modulo",
+        nfolds=3,
+        keep_cross_validation_predictions=True)
+
+    drf.train(data.x, data.y, data.train, validation_frame=data.train)
+
+    gs1 = H2OGridSearch(
+        H2OGradientBoostingEstimator(
+            fold_assignment="modulo",
+            nfolds=3,
+            keep_cross_validation_predictions=True),
+        hyper_params=hyper_parameters)
+
+    gs2 = H2OGridSearch(
+        H2ORandomForestEstimator(
+            fold_assignment="modulo",
+            nfolds=3,
+            keep_cross_validation_predictions=True),
+        hyper_params=hyper_parameters)
+
+    gs1.train(data.x, data.y, data.train, validation_frame=data.train)
+    gs2.train(data.x, data.y, data.train, validation_frame=data.train)
+
+    # List of model
+    # list of models will be then also the reference for the base models retrieved from the grid
+    se0 = H2OStackedEnsembleEstimator(base_models=[drf])
+    assert se0.base_models == [drf.model_id]
+
+    # Single Grid
+    se1 = H2OStackedEnsembleEstimator(base_models=gs1.models, seed=seed)
+    se1.train(data.x, data.y, data.train)
+
+    se2 = H2OStackedEnsembleEstimator(base_models=gs1, seed=seed)
+    se2.train(data.x, data.y, data.train)
+
+    assert sorted(se1.base_models) == sorted(se2.base_models)
+
+    # List of Grids
```

Comment:
if possible, create a test method for each use-case.
Python fails at first failed assertion, so status is more clear when all tests are executed: also tests run in isolation this way.
It's a bit different with R test runner, but still, 1 use-case = 1 test is necessary for proper isolation.